nixpkgs/nixpkgs.bzl

```diff
@@ -82,7 +88,17 @@ Set which Nixpkgs to use explicitly using 'repository' or 'path' attributes.
   # access to MAX_INT and 0 is not a valid timeout so this is as good
   # as we can do.
   timeout = 1073741824
-  res = ctx.execute(nix_build, quiet = False, timeout = timeout)
+
+  # Here we override NIX_PATH to something silly
+  # hence it will be impossible to use a implicit nixpkgs repository
+
+  # XXX: we have no way to ensure that the included 'nix_file' or
+  # 'nix_file_content' point to a pinned nixpkgs repository, so
+  # 'NIX_PATH' contains an error message, which will be seen by bazel user
+  res = ctx.execute(nix_build, quiet = False, timeout = timeout, environment=dict(
+    NIX_PATH="nixpkgs=implicit nixpkgs value is disabled for hermeticity reason."
```

Comment:
Why do this? Setting `NIX_PATH=` (i.e. to the empty string) works equally well.

Reply:
This is like using 0 as an error value instead of a Maybe type, or Verisign replacing all "Domain not found" errors with [Site Finder](https://en.wikipedia.org/wiki/Site_Finder). a) we should not be in the business of abusing environment variables meant for structured *input* to the Nix interpreter as unstructured *output* to the user. b) we should not assume that users can't follow a structured debugging process. Give them good documentation and get out of their way.

Reply:
Ok.

Reply:
I agree that the solution is hackish, but I rather prefer that than an error with limited details.